src/index.ts

```diff
@@ -89,26 +89,31 @@ export class Spectral {
     return rules;
   }
 
-  public setRules(rulesets: types.IRuleset[]) {
-    this._rulesets = merge([], rulesets);
-    this._functions = this._rulesetsToFunctions(this._rulesets);
-    this._rulesByIndex = this._rulesetsToRules(this._rulesets);
+  public setRules(
+    rulesets: types.IRuleset[],
+    opts: { includeCurrent: boolean } = { includeCurrent: true }
```

Comment:
See my comment in one of the test and let me know what you think about breaking that up into two separate, public methods:
- `updateRules`
- `setRules`

Reply:
Updated